samcli/lib/package/utils.py

```diff
@@ -284,3 +284,13 @@ def copy_to_temp_dir(filepath):
     dst = os.path.join(tmp_dir, os.path.basename(filepath))
     shutil.copyfile(filepath, dst)
     return tmp_dir
+
+
+def get_cf_template_name(temp_file, template_str, extension):
```

Comment:
Should probably be something like (using `typing.TextIO` for opened text-based file descriptors):

```suggestion
def get_cf_template_name(temp_file: TextIO, template_str: str, extension: str) -> str:
```